From da1b4af541bfef43c37c8bd15fb786944f2c4310 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20Br=C3=BCns?= <stefan.bruens@rwth-aachen.de>
Date: Sun, 19 Jan 2020 16:41:38 +0100
Subject: [PATCH 2/2] Add support for PR_CAPBSET_READ and _DROP syscalls

PR_CAPBSET_READ is used by the latest version of libcap (2.30), and
valgrind issues an error message arg4/arg5 are pointing to uninitialized
memory on ix86.
---
 coregrind/m_syswrap/syswrap-linux.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/coregrind/m_syswrap/syswrap-linux.c b/coregrind/m_syswrap/syswrap-linux.c
index 73ef98d..139e85d 100644
--- a/coregrind/m_syswrap/syswrap-linux.c
+++ b/coregrind/m_syswrap/syswrap-linux.c
@@ -1502,6 +1502,11 @@ PRE(sys_prctl)
       } else {
          PRE_REG_READ2(int, "prctl", int, option, int, mode);
       }
+   case VKI_PR_CAPBSET_READ:
+      PRE_REG_READ2(int, "prctl", int, option, int, capability);
+      break;
+   case VKI_PR_CAPBSET_DROP:
+      PRE_REG_READ2(int, "prctl", int, option, int, capability);
       break;
    default:
       PRE_REG_READ5(long, "prctl",
-- 
2.24.1

